src/web/lib/hakuneko/frontend@modern/components/menuComponent/script.js

```diff
@@ -0,0 +1,181 @@
+            { label: this.i18n('settings.colors.blue'), value: "blue", color: "#2196F3" },
+            { label: this.i18n('settings.colors.cian'), value: "cian", color: "#00BCD4" },
+            { label: this.i18n('settings.colors.teal'), value: "teal", color: "#009688" },
+            { label: this.i18n('settings.colors.green'), value: "green", color: "#4CAF50" },
+            { label: this.i18n('settings.colors.lgreen'), value: "lgreen", color: "#8BC34A" },
+            { label: this.i18n('settings.colors.lime'), value: "lime", color: "#CDDC39" },
+            { label: this.i18n('settings.colors.yellow'), value: "yellow", color: "#FFEB3B" },
+            { label: this.i18n('settings.colors.amber'), value: "amber", color: "#FFC107" },
+            { label: this.i18n('settings.colors.orange'), value: "orange", color: "#FF9800" },
+            { label: this.i18n('settings.colors.dorange'), value: "dorange", color: "#FF5722" },
+        ];
+
+        this.languages = [
+            { label: this.i18n('settings.languages.english'), value: 'en'},
+            { label: this.i18n('settings.languages.spanish'), value: 'es'},
+        ]
+    }
+
+    getLanguageSelected(value) {
+        return localStorage.getItem("lang") === value ? "selected" : "";
+    }
+
+    getCornerSelected(value) {
+        return localStorage.getItem("corner") === value ? "selected" : "";
+    }
+
+    getThemeSelected(theme) {
+        return localStorage.getItem("themeColor") === theme ? "selected" : "";
+    }
+
+    getPopupClass(visibility) {
+        return visibility ? "show" : "hide";
+    }
+
+    getPopupClassAppearance(visibility) {
+        return visibility ? "show" : "hide";
+    }
+
+    async togglePopup() {
+        let visible = !this.popupVisibility;
+        if (visible) {
+            this.set(
+                "settingCategories",
+                Engine.Settings.getCategorizedSettings()
+            );
+        } else {
+            this.set("settingCategories", []);
+        }
+        this.set("popupVisibility", visible);
+    }
+
+    async togglePopupAppearance() {
+        let visible = !this.popupVisibilityAppearance;
+        this.set("popupVisibilityAppearance", visible);
+    }
+
+    closeDiscardChanges() {
+        Engine.Settings.load();
+        this.set("settingCategories", []);
+        this.set("popupVisibility", false);
+    }
+
+    closeSaveChanges() {
+        Engine.Settings.save();
+        this.set("settingCategories", []);
+        this.set("popupVisibility", false);
+    }
+
+    setLanguage(e) {
+        localStorage.setItem("lang", e.model.item.value);
+        this.windowReload();
+    }
+
+    setTheme(e) {
+        localStorage.setItem("themeColor", e.model.item.value);
+        this.windowReload();
+    }
+
+    setCorner(e) {
+        localStorage.setItem("corner", e.model.item.value);
+        this.windowReload();
+    }
+
+    windowReload() {
+        window.location.reload();
```

Comment:
Use `confirm()` to show a dialog asking the user before reload, they may not want to do it now.